### PR TITLE
Stateless real-time mget

### PR DESCRIPTION
**Pull request by pxsalehi**
The mget counterpart of https://github.com/elastic/elasticsearch/pull/93976.

Relates ES-5677

### CODE REVIEW

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/mget/40_routing.yml [hunk 1]
```diff
@@ -8,11 +8,6 @@ routing:
             index:
               number_of_shards: 5
               number_of_routing_shards: 5
-              number_of_replicas: 0
```
Comment (pxsalehi) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/mget/40_routing.yml:11]: In Serverless, these tests need a search shard.

**Review by kingherc — COMMENTED** [review]
In general looks good to me. Just a couple of questions.

File: server/src/main/java/org/elasticsearch/action/get/TransportShardMultiGetAction.java [hunk 2]
```diff
@@ -163,6 +303,17 @@ private void asyncShardMultiGet(MultiGetShardRequest request, ShardId shardId, A
         }
     }
 
+    private DiscoveryNode getCurrentNodeOfPrimary(ShardId shardId) {
```
[the remark this answers is not in the document]
Comment (pxsalehi) [reply server/src/main/java/org/elasticsearch/action/get/TransportShardMultiGetAction.java]: Done in https://github.com/elastic/elasticsearch/pull/96763/commits/36d8892c1a56e75394491b19cea54bef634fa5a6 plus the other comments.

**Review by kingherc — APPROVED** [review]
Thanks, LGTM

**Review by henningandersen — COMMENTED** [review]
This looks good, left a few comments.

**Comment by pxsalehi** [comment]
@elasticmachine update branch

**Review by henningandersen — APPROVED** [review]
LGTM.

**Comment by pxsalehi** [comment]
@elasticmachine please run elasticsearch-ci/part-3
(failed test is https://github.com/elastic/elasticsearch/issues/96822)

**Pull request merged**